modules/a1MediaRtdProvider.js

```diff
@@ -0,0 +1,74 @@
+import { MODULE_TYPE_RTD } from '../src/activities/modules.js';
+import { loadExternalScript } from '../src/adloader.js';
+import { submodule } from '../src/hook.js';
+import { getStorageManager } from '../src/storageManager.js';
+import { isEmptyStr, mergeDeep } from '../src/utils.js';
+
+const REAL_TIME_MODULE = 'realTimeData';
+const MODULE_NAME = 'a1Media';
+const SCRIPT_URL = 'https://linkback.contentsfeed.com/src';
+export const A1_SEG_KEY = '__a1tg';
+export const A1_AUD_KEY = 'a1gid';
+
+export const storage = getStorageManager({moduleType: MODULE_TYPE_RTD, moduleName: MODULE_NAME});
+
+/** @type {RtdSubmodule} */
+export const subModuleObj = {
+  name: MODULE_NAME,
+  init: init,
+  getBidRequestData: alterBidRequests,
+};
+
+export function getStorageData(key) {
+  let storageValue = '';
+  if (storage.getDataFromLocalStorage(key)) {
+    storageValue = storage.getDataFromLocalStorage(key);
+  } else if (storage.getCookie(key)) {
+    storageValue = storage.getCookie(key);
+  }
+  return storageValue;
+}
+
+function loadLbScript(tagname) {
+  const linkback = window.linkback = window.linkback || {};
+  if (!linkback.l) {
+    linkback.l = true;
+
+    const d = new Date();
+    const month = `0${(d.getMonth() + 1)}`.slice(-2);
+    const day = `0${d.getDate()}`.slice(-2);
+    const todaysDate = `${d.getFullYear()}${month}${day}`;
+
+    const scriptUrl = `${SCRIPT_URL}/${todaysDate}/${tagname}.min.js`;
+    loadExternalScript(scriptUrl, MODULE_NAME);
+  }
+}
+
+function init(config, userConsent) {
+  const tagId = config.params.tagId;
+  if (tagId && !isEmptyStr(tagId)) {
+    loadLbScript(config.params.tagId);
+    return true;
+  }
+  if (!isEmptyStr(getStorageData(A1_SEG_KEY))) {
+    return true;
+  }
+  return false;
+}
+
+function alterBidRequests(reqBidsConfigObj, callback, config, userConsent) {
+  const a1seg = getStorageData(A1_SEG_KEY);
+  const a1gid = getStorageData(A1_AUD_KEY);
+  const a1Ortb2 = {
+    user: {
+      buyeruid: a1gid,
```

Comment:
This is the wrong place I'm sure; can you add some detail on this id and perhaps we can find the right place.

Reply:
It is a 1st party id that assigned to users by A1Media for the purpose of analysis and targeting.It is a 1st party user id.

Reply:
Please see the https://github.com/InteractiveAdvertisingBureau/openrtb/blob/master/extensions/2.x_official_extensions/eids.md spec.

Reply:
We moved from `user.buyeruid` to `user.ext.eids`